Should disable vue/html-indent within pre elements

By default the HTML 'pre' element contents are validated by the vue/html-indent rules.  I can work around this by manually adding pre to the ignores array in the rule definition, or by wrapping 'pre' in eslint-disable commands, but it seems 'pre' should be ignored by default.

**Tell us about your environment**

```
$ npm list | grep eslint
+-- babel-eslint@8.2.1
| +-- eslint-scope@3.7.1
| `-- eslint-visitor-keys@1.0.0
+-- eslint@4.16.0
| +-- eslint-scope@3.7.1 deduped
| +-- eslint-visitor-keys@1.0.0 deduped
+-- eslint-config-standard@11.0.0-beta.0
+-- eslint-friendly-formatter@3.0.0
+-- eslint-loader@1.9.0
+-- eslint-plugin-html@4.0.2
+-- eslint-plugin-import@2.8.0
| +-- eslint-import-resolver-node@0.3.2
| +-- eslint-module-utils@2.1.1
+-- eslint-plugin-node@5.2.1
+-- eslint-plugin-promise@3.6.0
+-- eslint-plugin-standard@3.0.1
+-- eslint-plugin-vue@4.2.0
| `-- vue-eslint-parser@2.0.2
|   +-- eslint-scope@3.7.1 deduped
|   +-- eslint-visitor-keys@1.0.0 deduped
| +-- @sheerun/eslint-config-standard@10.2.1
| +-- babel-eslint@8.2.1 deduped
| +-- eslint@4.16.0 deduped
| +-- prettier-eslint@8.8.1
| | +-- eslint@4.16.0 deduped
| | `-- typescript-eslint-parser@11.0.0

$ node --version
v8.9.0
```

**What did you do? Please include the actual source code causing the issue.**


```vue
<template>
  <pre>
some preformatted
text here
  </pre>
<template>
```

**What did you expect to happen?**

I expected no linter errors

**What actually happened? Please include the actual, raw output from ESLint.**

  44:1  error  Expected indentation of 10 spaces but found 2 spaces  vue/html-indent

--- a/lib/utils/indent-common.js
+++ b/lib/utils/indent-common.js
@@ -798,10 +798,12 @@ module.exports.defineVisitor = function create (context, tokenStore, defaultOpti
 
     VElement (node) {
       const startTagToken = tokenStore.getFirstToken(node)
-      const childTokens = node.children.map(n => tokenStore.getFirstToken(n))
       const endTagToken = node.endTag && tokenStore.getFirstToken(node.endTag)
 
-      setOffset(childTokens, 1, startTagToken)
+      if (node.name !== 'pre') {
+        const childTokens = node.children.map(n => tokenStore.getFirstToken(n))
+        setOffset(childTokens, 1, startTagToken)
+      }
       setOffset(endTagToken, 0, startTagToken)
     },
 

--- a/tests/lib/rules/html-indent.js
+++ b/tests/lib/rules/html-indent.js
@@ -230,6 +230,19 @@ tester.run('html-indent', rule, loadPatterns(
         // Ignore all :D
         ignores: ['*']
       }]
+    },
+
+    // Pre
+    {
+      code: unIndent`
+        <template>
+          <pre>
+        aaa
+          bbb
+            ccc
+          </pre>
+        </template>
+      `
     }
   ],
 
@@ -479,6 +492,38 @@ tester.run('html-indent', rule, loadPatterns(
         { message: 'Expected indentation of 8 spaces but found 6 spaces.', line: 3 },
         { message: 'Expected indentation of 12 spaces but found 8 spaces.', line: 4 },
         { message: 'Expected indentation of 12 spaces but found 8 spaces.', line: 6 }
+      ]
+    },
+
+    // Pre
+    {
+      code: unIndent`
+        <template>
+        <pre
+        style=""
+        >
+        aaa
+          bbb
+            ccc
+        </pre>
+        </template>
+      `,
+      output: unIndent`
+        <template>
+          <pre
+            style=""
+          >
+        aaa
+          bbb
+            ccc
+          </pre>
+        </template>
+      `,
+      errors: [
+        { message: 'Expected indentation of 2 spaces but found 0 spaces.', line: 2 },
+        { message: 'Expected indentation of 4 spaces but found 0 spaces.', line: 3 },
+        { message: 'Expected indentation of 2 spaces but found 0 spaces.', line: 4 },
+        { message: 'Expected indentation of 2 spaces but found 0 spaces.', line: 8 }
       ]
     }
   ]

